### PR TITLE
PHPStan 1.0 and Rector 0.12

### PR DESCRIPTION
**Pull request by mglaman**
## Description
Extends #184 with fixes to base classes.

I would recommend tagging 0.12.0 for this commit, following the release version pattern of Rector

## Drupal.org issue
https://www.drupal.org/project/rector/issues/3248684

### CODE REVIEW

File: src/Utility/TraitsByClassHelperTrait.php [hunk 1]
```diff
@@ -36,4 +38,16 @@ final protected function getTraitsByClass(string $class)
...
+    {
+        if ($class === null) {
+            return false;
+        }
+        $className = $this->nodeNameResolver->getName($class);
+        return in_array(
+            $trait,
+            $this->getTraitsByClass($className)
+        );
+    }
```
Comment (mglaman) [inline src/Utility/TraitsByClassHelperTrait.php:52]: Previously the code was just checking if the class name was an empty string. Now Rector leverages a type object lookup. This helps keep code clean when checking if a class node has a trait.

File: stubs/Drupal/Tests/UnitTestCase.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
```
Comment (mglaman) [inline stubs/Drupal/Tests/UnitTestCase.php:1]: Adds a stub so we can unit test the GetMockBase rector set.

File: tests/src/Rector/Deprecation/AssertPatternRector/AssertPatternRectorTest.php [hunk 3]
```diff
@@ -6,7 +6,7 @@
 use Rector\Testing\PHPUnit\AbstractRectorTestCase;
 use Symplify\SmartFileSystem\SmartFileInfo;
 
-class AssertHeaderRectorTest extends AbstractRectorTestCase {
+class AssertPatternRectorTest extends AbstractRectorTestCase {
```
Comment (mglaman) [inline tests/src/Rector/Deprecation/AssertPatternRector/AssertPatternRectorTest.php:9]: Fix for autoloading and Composer warning.

File: tests/src/Rector/Deprecation/GetMockRector/GetMockRectorTest.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+use Symplify\SmartFileSystem\SmartFileInfo;
+
+class GetMockRectorTest extends AbstractRectorTestCase {
+
+    /**
+     * @dataProvider provideData()
+     */
+    public function test(SmartFileInfo $fileInfo): void
+    {
+        $this->doTestFileInfo($fileInfo);
```
Comment (mglaman) [inline tests/src/Rector/Deprecation/GetMockRector/GetMockRectorTest.php:16]: This was one of the Rector rules crashing due to upstream changes. So I added a Unit test as well so we can more easily catch them without running the full functional test.

**Review by navneet0693 — APPROVED** [review]

**Review by FlorentTorregrosa — APPROVED** [review]
If with those changes, "require" section can be updated then it is ok for me. Thanks a lot!

**Pull request merged**